### PR TITLE
Ticket 10 get articles queries

### PR DESCRIPTION
**Pull request by JRGPower**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -57,6 +57,149 @@ describe('ENDPOINT TESTS', () => {
...
+                    return request(app)
+                        .get("/api/articles/?topic=mitch&sort_by=votes&order=asc")
+                        .expect(200)
+                        .then((res) => {
+                            expect(res.body.articles).toBeInstanceOf(Array)
+                            expect(res.body.articles.length).toBeGreaterThan(0)
+                            expect(res.body.articles).toBeSortedBy("votes", {
+                                ascending: true,
+                            });
+                        });
```
Comment (pkaodev) [inline __tests__/app.test.js]: this test currently isn't checking that the returned objects all have the topic of 'mitch'

File: __tests__/app.test.js [hunk 2]
```diff
@@ -57,6 +57,149 @@ describe('ENDPOINT TESTS', () => {
...
+                        .then((res) => {
+                            expect(res.body.msg).toBe("Bad Request");
+                        });
+                })
+                test("GET 400 - invalid topic", () => {
+                    return request(app)
+                        .get("/api/articles/topic&=chesss")
+                        .expect(400)
+                        .then((res) => {
+                            expect(res.body.msg).toBe("Bad Request");
```
Comment (pkaodev) [inline __tests__/app.test.js]: if the topic is non-existent it should be a 404

File: models/articles.model.js [hunk 3]
```diff
@@ -23,39 +23,64 @@ exports.checkExists = (colName, tableName, query, msg) => {
     }
 }
 
-exports.selectArticles = (articleId) => {
+exports.selectArticles = (articleId, topic, sort_by = "created_at", order = "desc") => {
```
Comment (pkaodev) [inline models/articles.model.js]: right now you are using this model for both selecting all articles and selecting articles by id.

split this into two models which handle those two cases separately.

**Comment by pkaodev** [comment]
Overall looking good.  Couple of things:
-make sure your topic test checks that all objects have the correct topic
-the test for non-existent topic should be a 404
-split the model you are using for all articles and article by id are separate.

please submit another PR with these changes.

**Comment by Geo-Beard** [comment]
Looks OK - feel free to merge and move on :)

**Pull request merged**